src/daos-deserialize/daos-deserialize.c

```diff
@@ -0,0 +1,225 @@
+     * be input on cmd line */
+    /* TODO: maybe support moving into multiple pools? */
+    /* TODO: broadcast pool handle to all ranks if not supporting
+     * multiple pools (instead of connecting on each rank)
+     * or have a check that if all pools are
+     * the same, then use the same handle */
+
+    /* create an empty file list */
+    mfu_flist tmplist = mfu_flist_new();
+
+    rc = daos_init();
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+        return 1;
+    }
+
+    /* use rank 0's paths, and spread them evenly among ranks
+     * Each "path" is a DAOS pool/cont, and this tool will
+     * serialize each container into its own hdf5 file */
+    int i;
+    if (rank == 0) {
+        for (i = 0; i < numpaths; i++) {
+                uint64_t idx = mfu_flist_file_create(tmplist);
+                mfu_flist_file_set_cont(tmplist, idx, argpaths[i]);
+        }
+    }
+
+    mfu_flist_summarize(tmplist);
+    mfu_flist newflist = mfu_flist_spread(tmplist);
+
+    /* get size of local list for each rank */
+    uint64_t size = mfu_flist_size(newflist);
+
+    daos_handle_t poh;
+    uuid_t cont_uuids[size];
+    daos_handle_t cohs[size];
+    bool daos_no_prefix = true;
+    daos_pool_info_t pool_info = {0};
+
+#if DAOS_API_VERSION_MAJOR < 1
+    rc = daos_pool_connect(pool_uuid, NULL, NULL, DAOS_PC_RW,
+                           &poh, &pool_info, NULL);
+#else
+    rc = daos_pool_connect(pool_uuid, NULL, DAOS_PC_RW,
+                           &poh, &pool_info, NULL);
+#endif
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to connect to pool");
+        rc = 1;
+    }
+
+    /* connect to each pool/cont in local list, then serialize */
+    for (i = 0; i < size; i++) {
+        char *path = mfu_flist_file_get_name(newflist, i);
+        uuid_generate(cont_uuids[i]);
+        printf("rank: %d, size: %d path: %s\n", rank, (int)size, path);
+
+
+        /* deserialize this hdf5 file to a DAOS container */
+        rc = cont_deserialize_hdlr(pool_uuid, cont_uuids[i],
+                                   poh, &cohs[i], path);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to deserialize container (%d)", rc);
+        }
+
+        rc = daos_cont_close(cohs[i], NULL);
+        MPI_Barrier(MPI_COMM_WORLD);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to close container (%d)", rc);
+            rc = 1;
+        }
+    }
+    
+    rc = daos_pool_disconnect(poh, NULL);
+    MPI_Barrier(MPI_COMM_WORLD);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to disconnect pool (%d)", rc);
+        rc = 1;
+    }
+
+    /* Alert the user if there were copy errors */
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "One or more errors were detected while "
+                "serializing: " MFU_ERRF, MFU_ERRP(MFU_ERR_DAOS));
+    }
+
+    rc = daos_fini();
+    if (rc != 0) {
+        rc = 1;
+    }
```

Comment:
@dsikich maybe log a message here so we don't return 1 from main without an indication of what went wrong.